pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -343,3 +349,45 @@ func (r *Cluster) ValidateDelete() error {
 
 	return nil
 }
+
+func validateServerVersionSkew(new, old *Cluster) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if old.IsSelfManaged() {
+		return allErrs
+	}
+
+	path := field.NewPath("spec")
+
+	oldVersion := old.Spec.KubernetesVersion
+	newVersion := new.Spec.KubernetesVersion
+
+	parsedOldVersion, err := version.ParseGeneric(string(oldVersion))
+	if err != nil {
+		allErrs = append(
+			allErrs,
+			field.Invalid(path, new.Spec.KubernetesVersion, fmt.Sprintf("parsing comparison version: %v", err.Error())))
+		return allErrs
+	}
+
```

Comment:
can we do the minor version difference check right after parsing it to keep related logic nearby?